src/world.cpp

```diff
@@ -2208,7 +2208,7 @@ void World_GenRoom(struct room_s *room, class VT_Level *tr)
     room->alternate_room = NULL;
     room->base_room = NULL;
 
-    if((tr_room->alternate_room >= 0) && ((uint32_t)tr_room->alternate_room < tr->rooms_count))
+    if((tr_room->alternate_room >= 0) && ((uint32_t)tr_room->alternate_room < tr->rooms_count) && (room->id < tr_room->alternate_room))
```

Comment:
Now I have a test map with three rooms. First room has the second as alternate, that has the third as alternate which has alternate room set to -1. All rooms has the same FLIPMAP, FLIPON and FLIPOFF triggers. But I can't cycle through these rooms, because the second alternate room get ignored and it swapped back to the first room.